kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -233,14 +248,7 @@ private void validate(Engine engine, SnapshotImpl snapshot, boolean isNewTable)
     // Validate the table has no features that Kernel doesn't yet support writing into it.
     TableFeatures.validateKernelCanWriteToTable(
         snapshot.getProtocol(), snapshot.getMetadata(), tablePath);
-
     if (!isNewTable) {
```

Comment:
should we perform the validation in the else block below regardless of `isNewTable` or not? or should we remove it completely from this fx and always do it during the other check only?

Reply:
We should do the validation in the else block regardless of if we're creating a new table or not. Currently the same validation is done in `validateUpdatedSchema` so we're covered but we can probably clean this up. I'd say we can do that in a follow on though, since this diff is already quite large and don't want to throw more refactorings